File: practice/algorithms/warmup/diagonal_difference/DiagonalDifference.js
function processData(input) {
    var lines = input.split('\n');
    var n = parseInt(lines[0]);
    var sumLeftRight = 0;
    var sumRightLeft = 0;
    for(var i = 0; i < n; i++)
    {
        var elements = lines[i + 1].split(" ").map(function(s) {return parseInt(s)});
        sumLeftRight += elements[i];
        sumRightLeft += elements[n - i - 1];
    }
    console.log(Math.abs(sumLeftRight - sumRightLeft));
}

process.stdin.resume();
process.stdin.setEncoding("ascii");
_input = "";
process.stdin.on("data", function (input) {
    _input += input;
});

process.stdin.on("end", function () {
    processData(_input);
});

File: practice/algorithms/warmup/diagonal_difference/DiagonalDifference.java
import java.util.Scanner;

public class DiagonalDifference 
{
	public static void main(String[] args)
	{
		Scanner stdin = new Scanner(System.in);
		int n = Integer.parseInt(stdin.nextLine());
		long sumLeftRight = 0;
		long sumRightLeft = 0;
		for(int i = 0; i < n; i++)
		{
			String[] elements = stdin.nextLine().split(" ");
			sumLeftRight += Long.parseLong(elements[i]);
			sumRightLeft += Long.parseLong(elements[n - i - 1]);
		}
		System.out.println(Math.abs(sumLeftRight - sumRightLeft));
		stdin.close();
	}
}
